File: Assets/Scripts/CSharp/CycleColour.cs
using UnityEngine;
using System.Collections;

public class CycleColour : MonoBehaviour 
{
	public Color startColor;
	public Color endColor;
	public float speed;

	void Start () 
	{		
		GetComponent<Renderer>().material = new Material( GetComponent<Renderer>().sharedMaterial );
	}
	
	void Update ()
	{ 
		float t = Mathf.PingPong( Time.time * speed, 1f );
		GetComponent<Renderer>().material.color = Color.Lerp( startColor, endColor, t );
	}
}

File: Assets/Scripts/Javascript/CycleColour.js
#pragma strict
/*
 * CycleColour
 */
var startColour : Color;
var endColour : Color;
var cycleSpeed : float;

function Start () {
	GetComponent.<Renderer>().material = new Material( GetComponent.<Renderer>().sharedMaterial );
}

function Update () {
	var time = Mathf.PingPong( Time.time * cycleSpeed, 1f );
	GetComponent.<Renderer>().material.color = Color.Lerp( startColour, endColour, time );
}
